src/library.js

```diff
@@ -3554,22 +3554,26 @@ addToLibrary({
 #endif
   },
 
+  // Sentinel for invalid handles; it's intentionally a distinct object so that
+  // we can distinguish it from `undefined` as an actual stored value.
+  $invalidHandleSentinel: {},
```

Comment:
Can we just use null instead maybe?

Reply:
Even better, we could use `this` and avoid and extra object, but not allow making a handle for the handle allocator itself

Reply:
Null isn't an option, because it is also a value stored

Reply:
Using `this` sounds good, with a comment explaining that its used.